transaction_receipt_query.go

```diff
@@ -271,17 +271,12 @@ func (query *TransactionReceiptQuery) Execute(client *Client) (TransactionReceip
 
 	if err != nil {
 		if precheckErr, ok := err.(ErrHederaPreCheckStatus); ok {
-			receipt := TransactionReceipt{}
-			if resp.query.GetTransactionGetReceipt() != nil {
-				receipt = _TransactionReceiptFromProtobuf(resp.query.GetTransactionGetReceipt().GetReceipt())
-			}
-
-			return receipt, ErrHederaReceiptStatus{
-				TxID:    precheckErr.TxID,
-				Status:  precheckErr.Status,
-				Receipt: receipt,
-			}
+			return TransactionReceipt{}, _NewErrHederaReceiptStatus(precheckErr.TxID, precheckErr.Status)
 		}
+		if respErr, ok := err.(ErrHederaReceiptStatus); ok {
```

Comment:
`_Execute` should never return an `ErrHederaReceiptStatus` and if it does, that would likely be the bug that needs to be fixed.

Reply:
Yes, this is happening.

Reply:
You can use the example I have provided to test it.

Reply:
To clarify I was saying we should probably fix that issue in this PR instead of checking for `ErrHederaReceiptStatus`, or at least we should make another PR that fixes that issue.

Reply:
Looked a bit at the code, fix _should(?)_ be as simple as changing `_TransactionReceiptQueryMapStatusError` to return `ErrHederaPreCheckStatus` instead of `ErrHederaReceiptStatus`

Reply:
Why would you want to return `ErrHederaPreCheckStatus` given that the error is not from this type?

Reply:
The reason is for the same reason you've mentioned before in some other comment .`TransactionReceiptQuery` itself should not error on anything except when `Header.NodeTransactionPrecheckCode` is an erring value; `Receipt.Status` should not be used to determine if `TransactionReceiptQuery` failed. However, we do want to return an error when someone uses the convince methods `Transaction[Response|ID].getReceipt()` as those add the extra requirement of `TransactionReceipt.Status` being `SUCCESS` else the receipt is determined to be an error. At least this is the expected behavior.